src/item_group.cpp

```diff
@@ -345,7 +345,7 @@ Item_spawn_data::ItemList Item_group::create(int birthday, RecursionList &rec) c
 
     for( auto& e : result ) {
         if( e.is_tool() || e.is_gun() || e.is_magazine() ) {
-            bool spawn_ammo = rng( 0, 99 ) < with_ammo && e.ammo_remaining() == 0;
+            bool spawn_ammo = rng( 0, 99 ) < with_ammo && e.ammo_remaining() == 0 && e.ammo_type() != "NULL";
```

Comment:
I think there are more occurrences of this idiom? Possibly the code for spawning items in vehicles is similarly affected?